arch/arc/core/reset.S

```diff
@@ -121,6 +121,35 @@ invalidate_dcache:
 
 done_cache_invalidate:
 
+/*
+ * ARC Custom Initialization
+ * Force Reset MPU (v3) registers
+ * For some test scenarios, this is needed to apply additional configurations
```

Comment:
Suggest to change/clarify: "For scenarios where the board hardware is not reset in between tests, some settings need to be reset in software"